toolkit/docs/building/building.md

```diff
@@ -1,248 +1,315 @@
+|Work on a package            | [Package Stage](#package-stage)                          |
+|Work on images               | [Image Stage](#image-stage)                              |
+|Work on the toolchain        | [Toolchain Stage](#toolchain-stage)                      |
+|Work on the tools            | [Tools](#tools)                                          |
+|Build the world from scratch | [Build From Scratch](#building-everything-from-scratch)  |
+
+### Install Prerequisites
+
+Prepare your system by [installing the necessary prerequisites.](prerequisites.md).
+
+### Clone and Sync to a Stable Commit
 
 Clone the 2.0-stable build of CBL-Mariner as shown here.
 
 ```bash
 # Get the source code
 git clone https://github.com/microsoft/CBL-Mariner.git
+
+# Make sure to fetch the latest tags
+git fetch --tags
+
 cd CBL-Mariner/toolkit
 
 # Checkout the desired release branch. The 2.0-stable tag tracks the most recent successful release of the 2.0 branch.
 git checkout 2.0-stable
 ```
 
-**IMPORTANT:** The 2.0-stable tag always points to the latest known good build of CBL-Mariner of the 2.0 branch. A similar tag, 1.0-stable, exists for the 1.0 branch. Other branchses are also buildable but not guarnateed to be stable.  The 1.0 and 2.0 branches are periodically updated with bug fixes, security vulnerability fixes or occasional feature enhancements.  As those fixes are integrated into the branch the head of a branch may be temporarily unstable.  The 2.0-stable tag will remain fixed until the tip of the branch is validated and the latest source and binary packages (SRPMs and RPMs) are published.  At that point, the 2.0-stable tag is advanced.  To ensure you have the latest invoke _git fetch --tags_ before building.
+**NOTE: All subsequent commands are assumed to be executed from inside the toolkit directory.**
 
-It is also possible to build an older version of CBL-Mariner from the 2.0 branch.  CBL-Mariner may be updated at any time, but an aggregate release is declared monthly and [tagged in github](https://github.com/microsoft/CBL-Mariner/releases).  These monthly builds are stable and their tags can be substituted for the 2.0-stable label above.
+### Stable Tags
 
-Alternate branches are not generally buildable because community builds require the SRPMs and/or RPMs be published.  At this time, published files are only available for the 2.0 branch.
+>_**IMPORTANT:**_ The `2.0-stable` tag always points to the last known good build of the `2.0` branch. \
+\
+A similar tag, `1.0-stable`, exists for the `1.0` branch. \
+\
+Other branches are also buildable but not guaranteed to be stable.  The `1.0` and `2.0` branches are periodically updated with bug fixes, security vulnerability fixes, or occasional feature enhancements.  As those fixes are integrated into the branch the head, of a branch may be temporarily unstable (ie.  only building using the full bootstrap build from source, rather than hydrating from the package servers). \
```

Comment:
```suggestion
Other branches are also buildable but not guaranteed to be stable.  The `1.0` and `2.0` branches are periodically updated with bug fixes, security vulnerability fixes, or occasional feature enhancements.  As those fixes are integrated into the branch, the head of a branch may be temporarily unstable (i.e., only building using the full bootstrap build from source, rather than hydrating from the package servers). \
```